pkg/verify/tsa.go

```diff
@@ -54,18 +54,26 @@ func VerifyTimestampAuthority(entity SignedEntity, trustedMaterial root.TrustedM
 	signatureBytes := sigContent.Signature()
 
 	verifiedTimestamps := []*root.Timestamp{}
+	var errs []error
 	for _, timestamp := range signedTimestamps {
 		verifiedSignedTimestamp, err := verifySignedTimestamp(timestamp, signatureBytes, trustedMaterial)
 
 		// Timestamps from unknown source are okay, but don't count as verified
 		if err != nil {
+			errs = append(errs, err)
 			continue
 		}
 
 		verifiedTimestamps = append(verifiedTimestamps, verifiedSignedTimestamp)
 	}
 
-	return verifiedTimestamps, nil
+	// Return errors if there are any, even if there are verified timestamps.
+	// This allows the caller to handle errors if the threshold is not met.
+	err = nil
+	if len(errs) > 0 {
+		err = fmt.Errorf("verifying timestamps: %w", errors.Join(errs...))
+	}
+	return verifiedTimestamps, err
```

Comment:
Like you mentioned, this is non conventional for Go. I'm partial to what you proposed in 473.